### PR TITLE
feat(docs): gate for private docs + compliance dashboard demo guide

### PR DESCRIPTION
**Pull request by esetnik**
## What

Adds a **reusable gate for non-public docs** and moves the compliance dashboard demo guide behind it.

- **`netlify/edge-functions/private-docs-auth.js`** — HTTP Basic Auth on any page under `/docs/private/*`. Drop a doc under `src/pages/docs/private/` and leave it out of the sidebar nav; it's automatically locked. No per-page wiring, no `netlify.toml` (the function declares its own path).
- **`/docs/private/compliance-dashboard-demo`** — the illustrated demo walkthrough (every shipped feature, screenshots from the seeded demo org). Gated because it carries an ephemeral demo link + seeded credentials, so it must not be public.

## Setup required before this is usable

Set two env vars on the Netlify site (all contexts), then the gate is live:

- `DOCS_PRIVATE_USER`
- `DOCS_PRIVATE_PASSWORD`

The gate **fails closed**: until these are set, `/docs/private/*` returns `401` (never leaks content). Preview and production share the one credential.

## Scope / caveats

- Protects the **rendered page** only. The Markdown source stays in this public repo, and page images under `/images/**` are **not** gated (anonymized UI screenshots, no secrets). Don't put real secrets in a private doc's committed source.
- `netlify/edge-functions/**` is excluded from eslint (Deno runtime globals).

## Verification

- `npm run build` ✓ — route `/docs/private/compliance-dashboard-demo` generated, Markdoc parses, all 11 `{% figure %}` tags valid.
- `eslint .` ✓
- Edge handler unit tests (Node, stubbed `Deno.env`): 13/13 passing; valid credentials and case-insensitive schemes → 200; missing/wrong credentials → 401; UTF-8 and colon-containing passwords handled; unset env → fail-closed 401; private index/data routes and no-store caching covered.
- **Please confirm on the deploy preview** after setting the env vars: un-authed `GET /docs/private/compliance-dashboard-demo` → 401; with credentials → 200.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Introduces access-control logic on the docs edge path (Basic Auth, fail-closed env handling, cache headers). Risk is contained to private docs routes and covered by unit tests, but misconfiguration or path gaps could leak or block gated content.
> 
> **Overview**
> Adds a **reusable HTTP Basic Auth gate** for any page under `/docs/private/*`, plus the first gated page: an internal Compliance Dashboard demo walkthrough.
> 
> A new Netlify edge function (`private-docs-auth.js`) checks `DOCS_PRIVATE_USER` / `DOCS_PRIVATE_PASSWORD`, **fails closed** when unset, covers both rendered pages and Next data routes, and forces `no-store` caching on authenticated responses. Dropping a Markdoc file under `src/pages/docs/private/` (and omitting it from sidebar nav) is enough to lock it—no per-page wiring.
> 
> Also wires unit tests into CI (`test:private-docs`), excludes Deno edge functions from eslint, and documents the private-docs pattern in `AGENTS.md`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3a6d303d45bc553cfcfa9b6aa2c31b97f31ab5f1. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: netlify/edge-functions/private-docs-auth.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  // Evaluate both comparisons regardless of the first result (no `&&`
+  // short-circuit) so a wrong username and a wrong password cost the same.
+  const userOk = safeEqual(user, expectedUser);
+  const passwordOk = safeEqual(password, expectedPassword);
+  if (!(userOk && passwordOk)) {
+    return unauthorized();
+  }
+
+  return context.next();
```
[the remark this answers is not in the document]
Comment (esetnik) [reply netlify/edge-functions/private-docs-auth.js]: [Written by Claude] Completed in 0ce43bc. Authenticated responses already set browser, CDN, and Netlify CDN cache controls to no-store; they now also append Authorization to Vary while preserving existing Vary tokens. The regression test verifies both behaviors.

File: netlify/edge-functions/private-docs-auth.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  if (!header.startsWith('Basic ')) {
+    return unauthorized();
+  }
+
+  let decoded;
+  try {
+    decoded = atob(header.slice('Basic '.length).trim());
+  } catch {
+    return unauthorized();
+  }
```
[the remark this answers is not in the document]
Comment (esetnik) [reply netlify/edge-functions/private-docs-auth.js:95]: [Written by Claude] Addressed at the current head: the handler now parses Authorization with a case-insensitive /^Basic\\s+/i expression, and the test suite covers mixed-case bAsIc.

File: test/private-docs-auth.test.mjs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  globalThis.Deno = { env: { get: () => undefined } }
+
+  const response = await handler(
+    new Request('https://docs.example.test/docs/private/guide'),
+    { next: () => new Response('must not be served') },
+  )
+
+  assert.equal(response.status, 401)
+  assert.equal(response.headers.get('cache-control'), 'no-store')
+})
```
[the remark this answers is not in the document]
Comment (esetnik) [reply test/private-docs-auth.test.mjs:210]: [Written by Claude] Addressed in ccdee06. The suite now covers a missing Authorization header with WWW-Authenticate, incorrect usernames and passwords, and passwords containing colons, matching the PR verification claims.

File: src/pages/docs/private/compliance-dashboard-demo.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+- "My tasks" vs "all tasks" views; the nav badge counts open tasks.
+- **Issues auto-complete when the underlying gap resolves**: fix the cert, and the task closes itself.
+
+Try it: click a gap cell in the Matrix to open its issue page, comment on it, then look for it in the Overview action queue.
+
+---
+
+## 13. Provider self-view ("My compliance")
+
+Log out and back in as **James Smith** (`proems-user-1@e2e.prodigyems.com`), or use the "View as Training Officer" toggle to see any provider's own dashboard:
```
[the remark this answers is not in the document]
Comment (esetnik) [reply src/pages/docs/private/compliance-dashboard-demo.md]: [Written by Claude] Addressed in 9f383c9. The committed guide no longer contains a demo URL, email address, username, or password; it directs readers to obtain rotating access details separately.

File: package.json [hunk 5]
```diff
@@ -6,7 +6,8 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "test:private-docs": "node --disable-warning=MODULE_TYPELESS_PACKAGE_JSON --test test/private-docs-auth.test.mjs"
```
[the remark this answers is not in the document]
Comment (esetnik) [reply package.json:10]: [Written by Claude] No code change is needed. This script names exactly one test file, and top-level node:test tests in that module run sequentially unless a test explicitly opts into concurrency. The suite never enables that option, so mutations of globalThis.Deno cannot overlap; the CLI test-concurrency setting controls scheduling across test files rather than making these tests concurrent.
